src/lib.rs

```diff
@@ -54,7 +62,37 @@ pub fn run(config: &Config) -> Result<(), Box<Error>> {
         Box::new(event_loop.handle()),
     );
 
-    observer::run(&origin, &auxiliary, &event_loop.handle());
+    let origin = Arc::new(origin);
+    let auxiliary = Arc::new(auxiliary);
+
+    // This will panic if construction will fail.
+    let contract_registry =
+        ContractRegistry::new(Arc::clone(&origin), Arc::clone(&auxiliary), config)
+            .expect("Error instantiating contract registry:");
+
+    let origin_reactors = reactor::origin_reactors(
+        Arc::clone(&origin),
+        Arc::clone(&auxiliary),
+        &contract_registry,
+        config,
+        Box::new(event_loop.handle()),
+    ).expect("Error instantiating origin reactors.");;
+
+    let auxiliary_reactors = reactor::auxiliary_reactors(
+        Arc::clone(&origin),
+        Arc::clone(&auxiliary),
+        &contract_registry,
+        config,
+        Box::new(event_loop.handle()),
+    ).expect("Error instantiating auxiliary reactors.");;
+
+    let origin_observer = Observer::new(origin, origin_reactors, Box::new(event_loop.handle()));
```

Comment:
What you usually do with handles is you `clone()` them and store them directly, not inside a `Box`: https://tokio-rs.github.io/tokio-core/tokio_core/reactor/struct.Core.html#method.handle